enterprise/proxysdk/client.go

```diff
@@ -0,0 +1,54 @@
+package proxysdk
+
+import (
+	"context"
+	"net/http"
+	"net/url"
+
+	"github.com/coder/coder/codersdk"
+)
+
+const (
+	// ExternalProxyTokenHeader is the auth header used for requests from
+	// external proxies.
+	//
+	// The format of an external proxy token is:
+	//     <proxy id>:<proxy secret>
+	//
+	//nolint:gosec
+	ExternalProxyTokenHeader = "Coder-External-Proxy-Token"
```

Comment:
Reminder that we can take `Proxy` out of this name since it's part of the package